pkg/doc/jwt/jwt.go

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package jwt
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const (
+	// HeaderB64 defines b64 header of boolean type
+	HeaderB64 = "b64" // bool
+
+	// HeaderType defines typ header of string type
+	HeaderType = "typ" // string
+
+	// HeaderAlgorithm defines alg header of string type
+	HeaderAlgorithm = "alg" // string
+
+	// HeaderKeyID defines kid header of string type
+	HeaderKeyID = "kid" // string
+
+	// TypeJWT defines JWT type
+	TypeJWT = "JWT"
+
+	// AlgorithmNone used to indicate unsecured JWT
+	AlgorithmNone = "none"
+
+	// SignatureEdDSA defines EdDSA alg
+	SignatureEdDSA = "EdDSA"
+
+	// SignatureRS256 defines RS256 alg
+	SignatureRS256 = "RS256"
+)
+
+const (
+	jwtPartsNumber   = 3
+	jwtHeaderPart    = 0
+	jwtPayloadPart   = 1
+	jwtSignaturePart = 2
+)
+
+// JSONWebToken defines JWT basic operations with JWT headers, payload and signature.
+type JSONWebToken interface {
+
+	// Claims fills input c with claims. It could be structure (e.g. Claims or its extension) or map[string]interface{}.
+	Claims(c interface{}) error
+
+	// VerificationData build verification data which can be used for signing or verification.
+	VerificationData() ([]byte, error)
+
+	// Header makes look up of particular header.
+	Header(name string) (interface{}, bool)
+
+	// HeaderString makes look up of particular header with string value.
+	HeaderString(name string) string
+
+	// Signature returns a signature if defined.
+	Signature() []byte
+
+	// CompactSerialize makes compact serialization of token.
+	CompactSerialize(detached bool) (string, error)
+}
+
+type jsonWebToken struct {
```

Comment:
It would be nice to make this generic between JWM and JWT.

Reply:
OK, it makes sense.